### PR TITLE
MES-1882: Handle HTTP 304 when fetching journals

### PR DESCRIPTION
**Pull request by johnmorleybjss**
## Description and relevant Jira numbers
* Journal service returns HTTP 304 when there are no changes as per https://github.com/dvsa/mes-journal-service/pull/28.
* Handle this gracefully by using the journal slots already cached on the device.
* Simplify how the `If-Modified-Since` header on the journal request is generated.

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [ ] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/providers/journal/journal.ts [hunk 1]
```diff
@@ -26,21 +26,21 @@ export class JournalProvider {
...
     private dateTimeProvider: DateTimeProvider,
-  ) {}
+  ) { }
 
   getJournal(lastRefreshed: Date): Observable<ExaminerWorkSchedule> {
     const staffNumber = this.authProvider.getEmployeeId();
     const journalUrl = this.urlProvider.getPersonalJournalUrl(staffNumber);
     const networkStatus = this.networkStateProvider.getNetworkState();
     if (lastRefreshed === null) {
       if (!this.authProvider.isInUnAuthenticatedMode() &&
```
Comment (MarkPatt89) [inline src/providers/journal/journal.ts]: At risk of being pernickety here, isInUnAuthenticatedMode, drop the upper case 'A'.

File: src/pages/journal/journal.effects.ts [hunk 2]
```diff
@@ -68,13 +68,28 @@ export class JournalEffects {
...
+              new journalActions.LoadJournalSuccess(
+                slots,
+                this.networkStateProvider.getNetworkState(),
+                this.authProvider.isInUnAuthenticatedMode(),
+                lastRefreshed,
+              ),
+            ),
+            catchError((err) => {
+              // For HTTP 304 NOT_MODIFIED we just use the slots we already have cached
+              if (err.status === 304) {
```
Comment (Cavalier07) [inline src/pages/journal/journal.effects.ts]: If we aren't modifying the data do we need to run the success action as the data will just be the same right

Comment (johnmorleybjss) [reply src/pages/journal/journal.effects.ts]: I tried this, but the spinner stayed up as that action is triggering the spinner dismissal.

Comment (Cavalier07) [reply src/pages/journal/journal.effects.ts]: Just looked and it seems a pain to untangle, doing it this way won't cause any issues :)

**Pull request merged**